File: medico/Domain/Medico.Domain/Models/Room.cs
using System;
using System.Collections.Generic;

namespace Medico.Domain.Models
{
    public class Room : Entity
    {
        public bool IsActive { get; set; }

        public Guid LocationId { get; set; }

        public Location Location { get; set; }

        public string Name { get; set; }

        public List<Appointment> Appointments { get; set; }
    }
}

File: client/src/dataModels/room.ts
import { BaseEntityActiveModel } from "./baseEntityModel";

export class Room extends BaseEntityActiveModel {
    LocationId: string;
    Name: string;

    constructor(id: string = "", isActive: boolean = true) {
        super(id, isActive);

        this.LocationId = "";
        this.Name = "";
    }

    createFromEntityModel(entity: any) {
        const newRoom = new Room(entity.Id, !!entity.IsActive);

        newRoom.LocationId = entity.LocationId;
        newRoom.Name = entity.Name;

        return newRoom;
    }
}